packages/protocol/lib/compatibility/ast-layout.ts

```diff
@@ -131,39 +158,51 @@ const compareStructDefinitions = (oldType: TypeInfo, newType: TypeInfo) => {
   }
 }
 
-const generateStructsCompatibilityReport = (oldLayout: StorageLayoutInfo, newLayout: StorageLayoutInfo) => {
+// Struct is expandable only if used in mappings or arrays
+const isStructExpandable = (oldType: TypeInfo, oldLayout: StorageLayoutInfo) => {
+  const structString = `t_struct<${oldType.label}>`
+  return !oldLayout.storage.some(storage => storage.type === structString)
+}
+
+const generateStructsCompatibilityReport = (oldLayout: StorageLayoutInfo, newLayout: StorageLayoutInfo): {compatible: boolean, errors: any[], expanded?: boolean} => {
   let compatible = true
   let errors = []
+  let expanded: boolean
```

Comment:
Is a single boolean signaling "some struct was expanded" that useful? If this info is relevant, wouldn't it be better to return this sort of info per struct?